lib/process-services-cloud/src/lib/task/task-filters/services/service-task-filter-cloud.service.ts

```diff
@@ -0,0 +1,306 @@
+        );
+    }
+
+    /**
+     * Deletes a task filter
+     * @param filter The filter to delete
+     * @returns Observable of task instance filters without deleted filter
+     */
+    deleteFilter(deletedFilter: ServiceTaskFilterCloudModel): Observable<ServiceTaskFilterCloudModel[]> {
+        const key = this.prepareKey(deletedFilter.appName);
+        return this.getTaskFiltersByKey(deletedFilter.appName, key).pipe(
+            switchMap((filters: any) => {
+                if (filters && filters.length > 0) {
+                    filters = filters.filter(filter => filter.id !== deletedFilter.id);
+                    return this.updateTaskFilters(deletedFilter.appName, key, filters);
+                }
+                return of([]);
+            }),
+            map(filters => {
+                this.addFiltersToStream(filters);
+                return filters;
+            }),
+            catchError((err) => this.handleTaskError(err))
+        );
+    }
+
+    /**
+     * Checks if given filter is a default filter
+     * @param filterName Name of the target task filter
+     * @returns Boolean value for whether the filter is a default filter
+     */
+    isDefaultFilter(filterName: string): boolean {
+        const defaultFilters = this.defaultServiceTaskFilters();
+        return defaultFilters.findIndex((filter) => filterName === filter.name) !== -1;
+    }
+
+    /**
+     * Calls update preference api to update task filter
+     * @param appName Name of the target app
+     * @param key Key of the task filters
+     * @param filters Details of update filter
+     * @returns Observable of updated task filters
+     */
+    private updateTaskFilters(appName: string, key: string, filters: ServiceTaskFilterCloudModel[]): Observable<ServiceTaskFilterCloudModel[]> {
+        return this.preferenceService.updatePreference(appName, key, filters);
+    }
+
+    /**
+     * Gets the username field from the access token.
+     * @returns Username string
+     */
+    getUsername(): string {
+        const user = this.identityUserService.getCurrentUserInfo();
+        return user.username;
+    }
+
+    /**
+     * Creates a uniq key with appName and username
+     * @param appName Name of the target app
+     * @returns String of task filters preference key
+     */
+    private prepareKey(appName: string): string {
+        return `task-filters-${appName}-${this.getUsername()}`;
+    }
+
+    /**
+     * Finds and returns the task filters from preferences
+     * @param appName Name of the target app
+     * @returns Array of TaskFilterCloudModel
+     */
+    private findFiltersByKeyInPreferences(preferences: any, key: string): ServiceTaskFilterCloudModel[] {
+        const result = preferences.find((filter: any) => { return filter.entry.key === key; });
+        return result && result.entry ? JSON.parse(result.entry.value) : [];
+    }
+
+    private handleTaskError(error: any) {
```

Comment:
remove this function as it does absolutely nothing, it rethrows the error, so you can just leave the function as it is with it's original error